src/include/rados/rados_types.hpp

```diff
@@ -327,5 +327,27 @@ struct inconsistent_snapset_t {
  */
 const std::string all_nspaces(LIBRADOS_ALL_NSPACES);
 
+struct notify_ack_t {
+  uint64_t notifier_id;
+  uint64_t cookie;
+  bufferlist payload_bl;
+
+  notify_ack_t(uint64_t notifier_id, uint64_t cookie,
+               bufferlist &payload)
+    : notifier_id(notifier_id),
+      cookie(cookie),
+      payload_bl(payload) {
+  }
+};
+
+struct notify_timeout_t {
+  uint64_t notifier_id;
+  uint64_t cookie;
+
+  notify_timeout_t(uint64_t notifier_id, uint64_t cookie)
+    : notifier_id(notifier_id),
+      cookie(cookie) {
+  }
```

Comment:
nit, this is identical to what we get with aggregate initialization. probably you could just drop it.

Reply:
ack